[rtaudio] build failure with mingw

**Host Environment**
- OS: Windows
- Compiler: mingw

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install rtaudio --triplet=x64-mingw-static`

**Failure logs** 
```
...
[10/20] C:\Qt\Tools\mingw810_64\bin\x86_64-w64-mingw32-g++.exe -DHAVE_GETTIMEOFDAY -DRTAUDIO_EXPORT -D__RTAUDIO_DEBUG__ -D__WINDOWS_WASAPI__ -Drtaudio_EXPORTS -IC:/Lichtsteuerung.git/vcpkg/buildtrees/rtaudio/src/5.1.0-9b49993cec.clean/include -IC:/Lichtsteuerung.git/vcpkg/buildtrees/rtaudio/src/5.1.0-9b49993cec.clean -Wall -Werror -g -MD -MT CMakeFiles/rtaudio.dir/RtAudio.cpp.obj -MF CMakeFiles\rtaudio.dir\RtAudio.cpp.obj.d -o CMakeFiles/rtaudio.dir/RtAudio.cpp.obj -c C:/Lichtsteuerung.git/vcpkg/buildtrees/rtaudio/src/5.1.0-9b49993cec.clean/RtAudio.cpp
FAILED: CMakeFiles/rtaudio.dir/RtAudio.cpp.obj 
C:\Qt\Tools\mingw810_64\bin\x86_64-w64-mingw32-g++.exe -DHAVE_GETTIMEOFDAY -DRTAUDIO_EXPORT -D__RTAUDIO_DEBUG__ -D__WINDOWS_WASAPI__ -Drtaudio_EXPORTS -IC:/Lichtsteuerung.git/vcpkg/buildtrees/rtaudio/src/5.1.0-9b49993cec.clean/include -IC:/Lichtsteuerung.git/vcpkg/buildtrees/rtaudio/src/5.1.0-9b49993cec.clean -Wall -Werror -g -MD -MT CMakeFiles/rtaudio.dir/RtAudio.cpp.obj -MF CMakeFiles\rtaudio.dir\RtAudio.cpp.obj.d -o CMakeFiles/rtaudio.dir/RtAudio.cpp.obj -c C:/Lichtsteuerung.git/vcpkg/buildtrees/rtaudio/src/5.1.0-9b49993cec.clean/RtAudio.cpp
C:/Lichtsteuerung.git/vcpkg/buildtrees/rtaudio/src/5.1.0-9b49993cec.clean/RtAudio.cpp:67:22: error: 'std::__cxx11::string convertCharPointerToStdString(const char*)' defined but not used [-Werror=unused-function]
   static std::string convertCharPointerToStdString(const char *text)
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus.exe: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

**Solution**
This was fixed [here](https://github.com/thestk/rtaudio/commit/110323c68bb83fd49993d1e24f1adf6892ed6999) in the code. Or you can change the compiler flags (as you can see in the commit).

--- a/ports/rtaudio/portfile.cmake
+++ b/ports/rtaudio/portfile.cmake
@@ -3,14 +3,21 @@ vcpkg_fail_port_install(ON_TARGET "UWP")
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO thestk/rtaudio
-    REF 5.1.0
-    SHA512 338a3a14cd4ea665ac7e94a275cb017bffd87cb10eb8ab6784fad320345ee828b8874439edd08c39efa48736edf9aa0622620784adc320473b03a8f81e17fff6
+    REF 34a3752e0c8249dc1780d196cd24e745425f0c77
+    SHA512 00fea107f409f6dc43154aaf69aeffa1a3385031778b5f7d1ae1cc8337ed4ab92a7917cc9eade848dedd746016b6eff6234088619cb8d6a9a3f26a63efde493e
     HEAD_REF master
 )
+
+if(VCPKG_CRT_LINKAGE STREQUAL static)
+    set(RTAUDIO_STATIC_MSVCRT ON)
+else()
+    set(RTAUDIO_STATIC_MSVCRT OFF)
+endif()
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
+    OPTIONS -DRTAUDIO_STATIC_MSVCRT=${RTAUDIO_STATIC_MSVCRT}
 )
 
 vcpkg_install_cmake()

--- a/ports/rtaudio/vcpkg.json
+++ b/ports/rtaudio/vcpkg.json
@@ -1,6 +1,6 @@
 {
   "name": "rtaudio",
-  "version-string": "5.1.0",
+  "version-date": "2021-01-25",
   "description": "A set of C++ classes that provide a common API for realtime audio input/output across Linux (native ALSA, JACK, PulseAudio and OSS), Macintosh OS X (CoreAudio and JACK), and Windows (DirectSound, ASIO and WASAPI) operating systems.",
   "homepage": "https://github.com/thestk/rtaudio",
   "supports": "!uwp"

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5185,7 +5185,7 @@
       "port-version": 1
     },
     "rtaudio": {
-      "baseline": "5.1.0",
+      "baseline": "2021-01-25",
       "port-version": 0
     },
     "rtlsdr": {

--- a/versions/r-/rtaudio.json
+++ b/versions/r-/rtaudio.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "088d877f612f5f77cb47a93b0854491baebb5178",
+      "version-date": "2021-01-25",
+      "port-version": 0
+    },
+    {
       "git-tree": "264f4c8d4aadd733d12dbf0d47b882c94f2e3f1f",
       "version-string": "5.1.0",
       "port-version": 0

